internal/sys/sys_test.go

```diff
@@ -110,12 +112,19 @@ func TestFileEntry_cachedStat(t *testing.T) {
 			ino, ft, err := f.CachedStat()
 			require.NoError(t, err)
 			require.Equal(t, fs.ModeDir, ft)
+			if !canReadDirInode() {
+				tc.expectedIno = 0
+			}
 			require.Equal(t, tc.expectedIno, ino)
 			require.Equal(t, &cachedStat{Ino: tc.expectedIno, Type: fs.ModeDir}, f.cachedStat)
 		})
 	}
 }
 
+func canReadDirInode() bool {
+	return runtime.GOOS != "windows" || strings.HasPrefix(runtime.Version(), "go1.20")
```

Comment:
Given that `runtime.GOOS != "windows"` is comp time const, I think separating these two conditions will teach Go compiler to optimize out the entire if block! 

```suggestion
	if runtime.GOOS != "windows" {
		return true
	} else {
		return strings.HasPrefix(runtime.Version(), "go1.20")
	}
```